rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/NodeRemover.java

```diff
@@ -0,0 +1,276 @@
+        hierarchyNodeSource.decrementTiles();
+        hierarchyHost.decrementTiles();
+        compactFlowPanel.remove(index);
+    }
+
+    private void removeDeployingNode() {
+        nodeIterator.remove();
+        hierarchyNodeSource.decrementTiles();
+        compactFlowPanel.remove(index);
+    }
+
+    /**
+     * Tries to find node source, if find, it sets fields dedicated to node source
+     * and make index point to this node source.
+     * @return true if node source was found
+     */
+    boolean findNodeSource() {
+        final Iterator<HierarchyNodeSource> nsIterator = compactFlowPanel.getModel().iterator();
+        while (nsIterator.hasNext()) {
+            final HierarchyNodeSource hierarchyNodeSource = nsIterator.next();
+            if (hierarchyNodeSource.getNodeSource().getSourceName().equals(sourceName)) {
+                this.hierarchyNodeSource = hierarchyNodeSource;
+                this.hierarchyNodeSourceIterator = nsIterator;
+                return true;
+            } else {
+                index += hierarchyNodeSource.getTilesNumber();
+            }
+        }
+        return false;
+    }
+
+    /**
+     * SHOULD be called after findNodeSource().
+     * Tries to find deploying node in the node source found before.
+     * @return true if deploying node was found
+     */
+    private boolean findDeploying() {
+        ++index;
+        final Iterator<NodeSource.Host.Node> nodeIterator = hierarchyNodeSource.getDeploying().iterator();
+        while (nodeIterator.hasNext()) {
+            final Node existing = nodeIterator.next();
+            if (existing.getNodeUrl().equals(nodeUrl)) {
+                this.nodeIterator = nodeIterator;
+                return true;
+            } else {
+                ++index;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * SHOULD be called after findNodeSource().
+     * Tries to find host in the node source found before.
+     * @return true if host was found
+     */
+    boolean findHost() {
+        ++index;
+        index += hierarchyNodeSource.getDeploying().size();
+        final Iterator<HierarchyHost> hostIterator = hierarchyNodeSource.getHosts().iterator();
+        while (hostIterator.hasNext()) {
+            final HierarchyHost hierarchyHost = hostIterator.next();
+            if (hierarchyHost.getHost().getHostName().equals(hostName)) {
+                this.hierarchyHost = hierarchyHost;
+                this.hierarchyHostIterator = hostIterator;
+                return true;
+            } else {
+                index += hierarchyHost.getTilesNumber();
+            }
+        }
+        return false;
+    }
+
+    /**
+     * SHOULD be called after findHost().
+     * Tries to find node in the host found before.
+     * @return true if node was found
+     */
+    boolean findNode() {
+        ++index;
+        final Iterator<NodeSource.Host.Node> nodeIterator = hierarchyHost.getNodes().iterator();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeIterator" which hides the field declared at line 53. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)